src/server/helpers/openidHelper.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (C) 2021 -present, NDLA
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { Issuer, generators } from 'openid-client';
+import { Request } from 'express';
+import config from '../../config';
+
+const handleConfigTypes = (
+  configVariable: string | boolean | undefined,
+): string => {
+  if (typeof configVariable === 'string') {
+    return configVariable;
+  }
+  return '';
+};
+
+const OPENID_DOMAIN =
+  'https://auth.dataporten.no/.well-known/openid-configuration';
+const FEIDE_CLIENT_ID = handleConfigTypes(config.feideClientID);
+const FEIDE_CLIENT_SECRET = handleConfigTypes(config.feideClientSecret);
+const LOGIN_REDIRECT_URI = `https://ndla-frontend.test.api.ndla.no/login/success`;
+const LOGOUT_REDIRECT_URI = `https://ndla-frontend.test.api.ndla.no/`;
+
+const getIssuer = async () => await Issuer.discover(OPENID_DOMAIN);
+
+const getClient = () =>
+  getIssuer().then(
+    issuer =>
+      new issuer.Client({
+        client_id: FEIDE_CLIENT_ID,
+        client_secret: FEIDE_CLIENT_SECRET,
+        redirect_uris: [LOGIN_REDIRECT_URI],
+        response_types: ['code'],
+      }),
+  );
+
+export const getRedirectUrl = async () => {
+  const code_verifier = generators.codeVerifier();
+  const code_challenge = generators.codeChallenge(code_verifier);
+
+  return getClient()
+    .then(async client =>
```

Comment:
Tror du må bestemme deg for 1 syntax-type, enten `async/await` eller `then()`. Herfra og nedover så er det en god blanding av begge, velg én

Reply:
Ble helt fokusert på authhelper, så glemte helt av denne filen. Herlich